staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -765,17 +765,31 @@ func validateCustomResourceDefinitionValidation(ctx context.Context, customResou
 
 var metaFields = sets.NewString("metadata", "kind", "apiVersion")
 
+type openAPISchemaErrorList struct {
+	schemaErrors field.ErrorList
+	celErrors    field.ErrorList
+}
+
+func (o openAPISchemaErrorList) appendErrors(list openAPISchemaErrorList) {
+	o.schemaErrors = append(o.schemaErrors, list.schemaErrors...)
+	o.celErrors = append(o.celErrors, list.celErrors...)
+}
+
+func (o openAPISchemaErrorList) allErrors() field.ErrorList {
+	return append(o.schemaErrors, o.celErrors...)
+}
+
 // ValidateCustomResourceDefinitionOpenAPISchema statically validates
-func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSchemaProps, fldPath *field.Path, ssv specStandardValidator, isRoot bool, opts *validationOptions, celContext *CELSchemaContext) field.ErrorList {
-	allErrs := field.ErrorList{}
+func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSchemaProps, fldPath *field.Path, ssv specStandardValidator, isRoot bool, opts *validationOptions, celContext *CELSchemaContext) openAPISchemaErrorList {
```

Comment:
nit: it's weird to have an exported method return a private type with only private fields and accessors

Reply:
Good catch. Fixed and godoc'd.